swarm/src/connection/pool.rs

```diff
@@ -513,11 +512,13 @@ where
             waker.wake();
         }
 
-        let connection = Connection::new(
+        let span = tracing::error_span!("connection", connection_id = %id, peer_id = %obtained_peer_id, peer_address = ?endpoint);
```

Comment:
I'd prefer to not debug-print the `endpoint` but instead call `get_remote_address` on it an display-print that one.